lib/elasticity/index_config.rb

```diff
@@ -0,0 +1,50 @@
+module Elasticity
+  class IndexConfig
+    ATTRS = [:index_base_name, :document_type, :mapping, :strategy].freeze
```

Comment:
I'm ok with making it optional as long as the library doesn't allow any write operations if it's not defined. Otherwise, it would cause problems of mismatching schema